src/containers/HomePage.js

```diff
@@ -1,30 +1,104 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { string } from 'prop-types';
+import { string, func } from 'prop-types';
+import { ToastContainer, ToastStore } from 'react-toasts';
+import { Route } from 'react-router-dom';
 
+import history from '../utils/history';
+import { editUser } from '../actions/sessionActions';
 import Map from '../components/common/Map';
 import Menu from '../components/common/Menu';
+import EditForm from '../components/user/EditForm';
+import TargetForm from '../components/user/TargetForm';
 import { MAPS_API } from '../constants/constants';
+import routes from '../constants/routesPaths';
 
-const HomePage = ({ userName }) => (
-  <div className="home-page">
-    <Menu user={userName} />
-    <Map
-      isMarkerShown
-      googleMapURL={MAPS_API}
-      loadingElement={<div className="maps-loading" />}
-      containerElement={<div className="maps-container" />}
-      mapElement={<div className="google-maps" />}
-    />
-  </div>
-);
+class HomePage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    navigator.geolocation.getCurrentPosition((position) => {
+      const { coords } = position;
+      this.setState({
+        coords
+      });
+    });
+  }
+
+  mapClick = ({ latLng }) => {
+    this.setState({
+      target: {
+        lat: latLng.lat(),
+        lng: latLng.lng(),
+      }
+    });
+    history.push(routes.newTarget);
+  }
+
+  eraseTarget = () => {
+    this.setState({
+      target: null
+    });
+  }
+
+  render() {
+    const { username, edit } = this.props;
+    return (
+      <div className="home-page">
+        <Route
```

Comment:
I see what you're doing by using Route here, but can you read and try to nest routes like it's done here?

https://reacttraining.com/react-router/web/example/route-config